### PR TITLE
feat(llc): added user blocking

### PR DESCRIPTION
**Pull request by deven98**
Added user blocking feature to the client

### CODE REVIEW

File: packages/stream_chat/lib/src/core/models/user_block.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    this.userId,
+    this.blockedUserId,
+    this.createdAt,
+  });
+
+  /// Create a new instance from a json
+  factory UserBlock.fromJson(Map<String, dynamic> json) =>
+      _$UserBlockFromJson(json);
+
+  /// Banned user.
```
Comment (xsahil03x) [inline packages/stream_chat/lib/src/core/models/user_block.dart:23]:
```suggestion
  /// Blocked user.
```

File: packages/stream_chat/lib/src/core/models/user_block.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  final User user;
+
+  /// User that banned the [user].
```
Comment (xsahil03x) [inline packages/stream_chat/lib/src/core/models/user_block.dart:26]:
```suggestion
  /// User that blocked the [user].
```

File: packages/stream_chat/lib/src/core/models/user_block.dart [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  final User? blockedUser;
+
+  /// Reason for the ban.
```
Comment (xsahil03x) [inline packages/stream_chat/lib/src/core/models/user_block.dart:29]:
```suggestion
  /// Reason for the block.
```

File: packages/stream_chat/lib/src/core/models/user_block.dart [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  final String? userId;
+
+  /// Reason for the ban.
+  final String? blockedUserId;
+
+  /// Timestamp when the [user] was banned.
```
Comment (xsahil03x) [inline packages/stream_chat/lib/src/core/models/user_block.dart:35]:
```suggestion
  /// Timestamp when the [user] was blocked.
```

File: packages/stream_chat/lib/src/core/models/user_block.dart [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  /// User that blocked the [blockedUser].
+  final User user;
+
+  /// User that was blocked by the [user].
+  final User? blockedUser;
+
+  /// Reason for the block.
+  final String? userId;
+
+  /// Reason for the ban.
```
Comment (xsahil03x) [inline packages/stream_chat/lib/src/core/models/user_block.dart:32]:
```suggestion
  /// Id of the [blockedUser].
```

File: packages/stream_chat/lib/src/core/models/user_block.dart [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  factory UserBlock.fromJson(Map<String, dynamic> json) =>
+      _$UserBlockFromJson(json);
+
+  /// User that blocked the [blockedUser].
+  final User user;
+
+  /// User that was blocked by the [user].
+  final User? blockedUser;
+
+  /// Reason for the block.
```
Comment (xsahil03x) [inline packages/stream_chat/lib/src/core/models/user_block.dart:29]:
```suggestion
  /// ID of the [user].
```

File: packages/stream_chat/lib/src/core/models/user_block.dart [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:stream_chat/src/core/models/user.dart';
+
+part 'user_block.g.dart';
+
+/// Contains information about a [User] that was blocked from a [Channel] or App.
```
Comment (deven98) [inline packages/stream_chat/lib/src/core/models/user_block.dart:7]:
```suggestion
/// Contains information about a [User] blocked from a [Channel] or App.
```

**Review by xsahil03x — APPROVED** [review]
LGTM, but let's add some extra coverage before merging

**Review by xsahil03x — APPROVED** [review]
LGTM

**Review by xsahil03x — APPROVED** [review]
LGTM

**Pull request merged**